ee/tables/tablewrapper/tablewrapper.go

```diff
@@ -0,0 +1,67 @@
+package tablewrapper
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/kolide/launcher/ee/gowrapper"
+	"github.com/kolide/launcher/pkg/traces"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+const DefaultTableTimeout = 2 * time.Minute
+
+type generateResult struct {
+	rows []map[string]string
+	err  error
+}
+
+// NewTablePluginWithTimeout returns a table plugin using the default table timeout.
+func NewTablePluginWithTimeout(slogger *slog.Logger, name string, columns []table.ColumnDefinition, gen table.GenerateFunc) *table.Plugin {
+	return NewTablePluginWithCustomTimeout(slogger, name, columns, gen, DefaultTableTimeout)
+}
+
+// NewTablePluginWithCustomTimeout returns a table plugin that will attempt to execute a query
+// up until the given timeout, at which point it will instead return no rows and a timeout error.
+func NewTablePluginWithCustomTimeout(slogger *slog.Logger, name string, columns []table.ColumnDefinition, gen table.GenerateFunc, genTimeout time.Duration) *table.Plugin {
+	wrappedGen := func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+		ctx, span := traces.StartSpan(ctx, "table_name", name)
+		defer span.End()
+
+		ctx, cancel := context.WithTimeout(ctx, genTimeout)
+		defer cancel()
+
+		// Kick off running the query
+		resultChan := make(chan *generateResult)
```

Comment:
I love that this has the effect is interrupting no matter what's happening. But I'm also slightly leery of these piling up. Is there any way to clean this up? I assume not. So maybe we need to add something only spawn one generate (per table) at a time?

Reply:
Discussed offline -- I will add some limits in a follow-up PR